apiserver/facades/client/application/deploy.go

```diff
@@ -225,7 +228,14 @@ func (api *APIBase) addUnits(
 		if err := saveMachineInfo(ctx, api.machineService, id); err != nil {
 			return nil, errors.Annotatef(err, "saving assigned machine %q for unit: %q", id, unit.Name())
 		}
+		machineToUnitMap[id] = append(machineToUnitMap[id], unitName)
+	}
+
+	// Assign units to machines via net nodes.
+	if err := api.stubService.AssignUnitsToMachines(ctx, machineToUnitMap); err != nil {
+		return nil, internalerrors.Errorf("assigning units to machines: %w", err)
```

Comment:
Will there be a mapping of the domain errors to the apiserver ones?

Reply:
I'm following the same pattern used elsewhere, so I assume so? Anyway, this is just a temporary call so doesn't matter too much.

Reply:
Well I ask because you could be returning a `MachineNotFound` from the stub service and the client consuming the api should react differently maybe.